src/services/PromotionService.ts

```diff
@@ -50,29 +50,25 @@ export async function getPromotions(query?: PromotionDTO[]): Promise<Promotion[]
     });
 }
 
-export async function getRestaurant({ id }: Promotion): Promise<Restaurant> {
+export async function getRestaurant(id: string): Promise<Restaurant> {
   // TODO: find restaurant given placeId from promotion
   return {
```

Comment:
Do we have a ticket to track this TODO?

Reply:
And also this would be the area where we could store the restaurant details in the session data? So that for the same user, if they click on the marker again, we don't need to call google places API again right

Reply:
Oh yeah, I'll link it in the code: https://promopal.atlassian.net/browse/PP-25

And I think that caching might already be done in the `GooglePlacesService`! This function would eventually call `GooglePlacesService.getRestaurantDetails` which checks the cache if it already has the details for this restaurant

Reply:
Oh wait you're right sorry forgot that we cache it.

Reply:
I opened this again since we've changed it now 😅  so you might need to add more details for user session?

Reply:
Ahhh thanks! Just opened up a ticket in the backlog for this: https://promopal.atlassian.net/browse/PP-84

I'm thinking this could be an improvement? When we fetch the restaurant, I think it should be attached to a promotion on the FE so that we won't need to make the additional request again (we'd just do `promotion.restaurant` if we need it again) but I think the ticket will be nice as a reminder to check that we're not making the additional requests!

Reply:
Oooh gotcha! Would that be enough for user state- so same idea as the hashmap we had previously?

Reply:
Actually yeah you're right, might be better to do it as the hashmap like you had previously so then it's accessible throughout the whole app in case we need it 😛